frontend/src/components/utils/NavBar/index.tsx

```diff
@@ -141,6 +161,14 @@ const NavBar = ({ headersData }: Props): ReactElement => {
     setDrawerOpen(false);
   }, [location]);
 
+  useEffect(() => {
+    if (user) {
+      setButtonText('Sign Out');
+    } else {
```

Comment:
For this part, please use conditional expression, like `setButtonText(!user ? 'Sign In' : 'Sign Out')`